src/config.py

```diff
@@ -185,17 +218,31 @@ def server_properties(self) -> List[str]:
         Returns:
             List of properties to be set
         """
-        return (
-            [
-                f"offsets.retention.minutes={self.charm.config['offsets-retention-minutes']}",
-                f"log.retention.hours={self.charm.config['log-retention-hours']}",
-                f"auto.create.topics={self.charm.config['auto-create-topics']}",
-                f"super.users={self.super_users}",
+        properties = [
+            f"offsets.retention.minutes={self.charm.config['offsets-retention-minutes']}",
+            f"log.retention.hours={self.charm.config['log-retention-hours']}",
+            f"auto.create.topics={self.charm.config['auto-create-topics']}",
+            f"super.users={self.super_users}",
+        ] + self.default_replication_properties + self.auth_properties + DEFAULT_CONFIG_OPTIONS.split("\n")
+
+        if self.charm.tls.enabled:
+            # Remove Non-SSL specific ports and config
+            properties = [
+                p
+                for p in properties
+                if not p.startswith(
+                    (
+                        "listeners=",
+                        "advertised.listeners=",
+                        "listener.name.sasl_plaintext.",
+                        "security.inter.broker.protocol=",
+                        "security.protocol=",
+                    )
+                )
             ]
```

Comment:
This might be harder to maintain. Maybe let's move the non-tls properties out of the GLOBAL at the top, and set them here in an `else:` instead.